ng-dev/auth/shared/firebase.ts

```diff
@@ -0,0 +1,36 @@
+import {
+  GoogleAuthProvider,
+  getAuth,
+  signInWithCredential,
+  linkWithCredential,
+  GithubAuthProvider,
+} from 'firebase/auth';
+import {
+  deviceCodeOAuthDance,
+  authorizationCodeOAuthDance,
+  GithubOAuthDanceConfig,
+  GoogleOAuthDanceConfig,
+} from './oauth.js';
+
+export async function loginToFirebase() {
+  /** The type of OAuth dance to do based on whether a session display is available. */
+  const oAuthDance = process.env.DISPLAY ? deviceCodeOAuthDance : authorizationCodeOAuthDance;
```

Comment:
Should this be swapped? i.e. device code is without a display, vs. the authorization code grant is for users with a webserver/and the redirect

see: e.g. https://oauth.net/2/grant-types/authorization-code/#:~:text=After%20the%20user%20returns%20to%20the%20client%20via%20the%20redirect%20URL and e.g. https://developers.google.com/identity/protocols/oauth2/limited-input-device#step-1:-request-device-and-user-codes

Reply:
whoops, swapped it in local testing and forgot to swap back.